app/background.cc

```diff
@@ -79,52 +78,56 @@ static void read_thread(background_t *pbkd) {
     if (pbkd->dbg) fprintf(stderr, "background: thread stop\n");
 }
 
-void *load_background(const char *path, int debug) {
-    background_t *pbkd = new background_t;
-    background_t &bkd = *pbkd;
-
-    bkd.dbg = debug;
-    bkd.cap.open(path, cv::CAP_ANY);    // explicitly ask for auto-detection of backend
-    if (!bkd.cap.isOpened()) {
-        if (bkd.dbg) fprintf(stderr, "cv::VideoCapture cannot open: %s\n", path);
-        delete pbkd;
-        return nullptr;
-    }
-    bkd.cap.set(cv::CAP_PROP_CONVERT_RGB, true);
-    int fcc = (int)bkd.cap.get(cv::CAP_PROP_FOURCC);
-    bkd.fps = bkd.cap.get(cv::CAP_PROP_FPS);
-    int cnt = (int)bkd.cap.get(cv::CAP_PROP_FRAME_COUNT);
-    // Here be the logic...
-    //  if: can read 2 video frames => it's a video
-    //  else: is loaded as an image => it's an image
-    //  else: it's not usable.
-    if (bkd.cap.read(bkd.raw) && bkd.cap.read(bkd.raw)) {
-        // it's a video, try a reset and start reader thread..
-        if (bkd.cap.set(cv::CAP_PROP_POS_FRAMES, 0))
-            bkd.frm = 0;
-        else
-            bkd.frm = 2;
-        bkd.vid = true;
-        bkd.run = true;
-        bkd.thr = std::thread(read_thread, pbkd);
-    } else {
-        // static image file, try loading..
-        bkd.cap.release();
-        bkd.raw = cv::imread(path);
-        bkd.vid = false;
-        if (bkd.raw.empty()) {
-            if (bkd.dbg) fprintf(stderr, "cv::imread cannot read: %s\n", path);
-            delete pbkd;
+std::shared_ptr<background_t> load_background(const char *path, int debug) {
+    std::shared_ptr<background_t> pbkd(new background_t);
```

Comment:
```suggestion
    std::shared_ptr<background_t> pbkd = std::make_shared<background_t>();
```

Even might:
```suggestion
    auto pbkd = std::make_shared<background_t>();
```

As the return type of `std::make_shared<T>` is `std::shared_ptr<T>` …

Also note the proposed dtor trick I commented yesterday.

Reply:
Ah - I missed the dtor trick, that makes sense now I've re-read it!